src/main.rs

```diff
@@ -88,23 +95,18 @@ pub fn main() {
         exit(EX_OK);
     }
 
-    // Configure debug output:
-    if options.flags.debug {
-        std::env::set_var("RUST_BACKTRACE", "1");
-    }
-
     // Locate the given subcommand:
     let Command::External(command) = &options.command.unwrap();
-    let Some(command_path) = find_external_subcommand("asimov-", &command[0]) else {
+    let Some(cmd_to_execute) = commands.find(&command[0]) else {
```

Comment:
You have here inadvertently fundamentally worsened the time complexity and performance of subcommand execution:
- Previously, it only required looking up a specific filename from directories in `PATH`; exactly as a shell would do it as well
- Now it requires scanning through every file entry in every directory in `PATH`, allocating memory to hold every found `asimov-` binary, and then again scanning through the resulting list of binaries to match the one to execute